ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockCreator.java

```diff
@@ -22,12 +23,31 @@
 import java.util.Optional;
 
 public interface BlockCreator {
-  Block createBlock(final long timestamp);
+  class BlockCreationResult {
+    private final Block block;
+    private final TransactionSelectionResults transactionSelectionResults;
```

Comment:
i love the overall concept for this addition, can we give this field a better name? if it is always going to be used for removed transactions, lets name it something that indicates those semantics. As is it implies these are the results of the transactions included in the block, which is misleading.

Reply:
> As is it implies these are the results of the transactions included in the block, which is misleading.

This field contains the transactions that were included in the block as well. This is a class that already exited in Besu and was extended with `List<TransactionValidationResult> invalidTransactions` field, now containing 

```java
public static class TransactionSelectionResults {
    private final List<Transaction> transactions = Lists.newArrayList();
    private final List<TransactionReceipt> receipts = Lists.newArrayList();
    private final List<TransactionValidationResult> invalidTransactions = Lists.newArrayList();
    private long cumulativeGasUsed = 0;
    // ... 
}
```

Anyway,  if you have a suggestion for a better name happy to change :)

Reply:
No, I think this is good, it's just a layer deeper than I looked. No problems with that.